src/crypto/chacha20.h

```diff
@@ -39,4 +45,48 @@ class ChaCha20
     void Crypt(const unsigned char* input, unsigned char* output, size_t bytes);
 };
 
+class FSChaCha20
+{
+private:
+    ChaCha20 c20;
+    size_t rekey_interval;
+    uint32_t messages_with_key;
```

Comment:
9d9ffc4: naming it as a counter would be more intuitive.
```suggestion
    uint32_t message_counter;
```

Reply:
Fixed.